testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/core/spi/ResourceClassSeparatorAnnotationTest.java

```diff
@@ -0,0 +1,108 @@
+import org.jboss.resteasy.test.core.spi.resource.ResourceClassSeparatorAnnotationAsFieldTargetEndPoint;
+import org.jboss.resteasy.test.core.spi.resource.ResourceClassSeparatorAnnotationAsMethodTargetEndPoint;
+import org.jboss.resteasy.test.core.spi.resource.ResourceClassSeparatorAnnotationAsParameterTargetEndPoint;
+import org.jboss.resteasy.test.core.spi.resource.ResourceClassSeparatorAnnotationBeanParam;
+import org.jboss.resteasy.utils.PermissionUtil;
+import org.jboss.resteasy.utils.PortProviderUtil;
+import org.jboss.resteasy.utils.TestUtil;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(Arquillian.class)
+public class ResourceClassSeparatorAnnotationTest {
+    static ResteasyClient client;
+
+    @Deployment
+    public static Archive<?> deploy() {
+        WebArchive war = TestUtil.prepareArchive(ResourceClassSeparatorAnnotationTest.class.getSimpleName())
+                .addClass(ResourceClassSeparatorAnnotationTest.class)
+                .addClass(PortProviderUtil.class)
+                .addAsManifestResource(PermissionUtil.createPermissionsXmlAsset(
+                        new SocketPermission(PortProviderUtil.getHost(), "connect,resolve"),
+                        new PropertyPermission("org.jboss.resteasy.port", "read"),
+                        new PropertyPermission("quarkus.tester", "read"),
+                        new RuntimePermission("getenv.RESTEASY_PORT"),
+                        new PropertyPermission("ipv6", "read"),
+                        new PropertyPermission("node", "read"),
+                        new PropertyPermission("arquillian.*", "read"),
+                        new RuntimePermission("accessDeclaredMembers"),
+                        new ReflectPermission("suppressAccessChecks")), "permissions.xml");
+        return TestUtil.finishContainerPrepare(war, null,
+                ResourceClassSeparatorAnnotationAsParameterTargetEndPoint.class,
+                ResourceClassSeparatorAnnotationAsFieldTargetEndPoint.class,
+                ResourceClassSeparatorAnnotationAsMethodTargetEndPoint.class,
+                ResourceClassSeparatorAnnotationBeanParam.class);
+    }
+
+    private String generateURL(String path) {
+        return PortProviderUtil.generateURL(path, ResourceClassSeparatorAnnotationTest.class.getSimpleName());
+    }
+
+    @Test
+    @RunAsClient
+    public void testSeparatorAsParameterTarget() throws IOException {
+        client = (ResteasyClient) ClientBuilder.newClient();
+
+        Response response = client
+                .target(generateURL("/separator/parameter/1,2,3"))
+                .request()
+                .get();
+
+        Assert.assertEquals(HttpResponseCodes.SC_OK, response.getStatus());
+        Assert.assertEquals("This is your sentence:123", IOUtils.toString((InputStream) response.getEntity()));
```

Comment:
IMO we shouldn't use the `IOUtils` here as at some point I'd like to remove the commons-io usage. ou could simply do `response.readEntity(String.class)`.